### ISSUE TITLE
ceph: cluster health checking invalid pointer dereference

### ISSUE THREAD

**Issue by BlaineEXE** [rook/rook#5826]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

The upgrade test upgrading from Rook v1.3.1 to master fails with the Operator log message
```
2020-07-14 16:49:06.159040 I | op-config: CephCluster "upgrade-ns" status: "Progressing". "Cluster is checking if updates are needed"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x58 pc=0x18677eb]

goroutine 705 [running]:
github.com/rook/rook/pkg/operator/ceph/cluster/mon.(*Cluster).checkHealth(0xc0010e3400, 0x0, 0x0)
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/cluster/mon/health.go:104 +0xab
github.com/rook/rook/pkg/operator/ceph/cluster/mon.(*HealthChecker).Check(0xc000f5b720, 0xc000cc8cc0)
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/cluster/mon/health.go:92 +0x143
created by github.com/rook/rook/pkg/operator/ceph/cluster.(*ClusterController).startMonitoringCheck
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/cluster/monitoring.go:116 +0x683

```
(see in https://jenkins.rook.io/job/rook/job/rook/job/PR-4916/37/artifact/_output/tests/TestCephUpgradeSuitepoolcheck_aws_1.17.x_upgrade-ns-system_rook-ceph-operator-5cfb779bdb-9dz72_1594745381.log)

**Expected behavior:**
It appears the operator may be attempting to start the health checker before `c.ClusterInfo` is set in `pkg/operator/ceph/cluster/mon/health.go` line `116` (`logger.Debugf("Checking health for mons in cluster. %s", c.ClusterInfo.Name)`)

**How to reproduce it (minimal and precise):**
Reproduces in at least some upgrade tests.

**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary
  * from upgrade test
* Operator's logs, if necessary
```

-----------------------------------------
CONTAINER: rook-ceph-operator
-----------------------------------------
2020-07-14 16:48:50.861953 I | rookcmd: starting Rook v1.3.0-beta.0.542.g7117fc1 with arguments '/usr/local/bin/rook ceph operator'
2020-07-14 16:48:50.862030 I | rookcmd: flag values: --add_dir_header=false, --alsologtostderr=false, --csi-cephfs-plugin-template-path=/etc/ceph-csi/cephfs/csi-cephfsplugin.yaml, --csi-cephfs-provisioner-dep-template-path=/etc/ceph-csi/cephfs/csi-cephfsplugin-provisioner-dep.yaml, --csi-cephfs-provisioner-sts-template-path=/etc/ceph-csi/cephfs/csi-cephfsplugin-provisioner-sts.yaml, --csi-rbd-plugin-template-path=/etc/ceph-csi/rbd/csi-rbdplugin.yaml, --csi-rbd-provisioner-dep-template-path=/etc/ceph-csi/rbd/csi-rbdplugin-provisioner-dep.yaml, --csi-rbd-provisioner-sts-template-path=/etc/ceph-csi/rbd/csi-rbdplugin-provisioner-sts.yaml, --enable-discovery-daemon=true, --enable-flex-driver=true, --enable-machine-disruption-budget=false, --help=false, --kubeconfig=, --log-flush-frequency=5s, --log-level=INFO, --log_backtrace_at=:0, --log_dir=, --log_file=, --log_file_max_size=1800, --logtostderr=true, --master=, --mon-healthcheck-interval=10s, --mon-out-timeout=15s, --operator-image=, --service-account=, --skip_headers=false, --skip_log_headers=false, --stderrthreshold=2, --v=0, --vmodule=
2020-07-14 16:48:50.862036 I | cephcmd: starting Rook-Ceph operator
2020-07-14 16:48:51.023676 I | cephcmd: base ceph version inside the rook operator image is "ceph version 15.2.4 (7447c15c6ff58d7fce91843b705a268a1917325c) octopus (stable)"
2020-07-14 16:48:51.042911 I | op-discover: rook-discover daemonset already exists, updating ...
2020-07-14 16:48:51.049307 I | operator: looking for secret "rook-ceph-admission-controller"
2020-07-14 16:48:51.076319 I | operator: creating admission controller pods
2020-07-14 16:48:51.096021 I | operator: rook-provisioner "ceph.rook.io/block" started using "ceph.rook.io" flex vendor dir
I0714 16:48:51.096278       6 leaderelection.go:242] attempting to acquire leader lease  upgrade-ns-system/ceph.rook.io-block...
2020-07-14 16:48:51.096746 I | operator: rook-provisioner "rook.io/block" started using "rook.io" flex vendor dir
2020-07-14 16:48:51.096768 I | operator: watching all namespaces for ceph cluster CRs
2020-07-14 16:48:51.097047 I | operator: setting up the controller-runtime manager
I0714 16:48:51.097599       6 leaderelection.go:242] attempting to acquire leader lease  upgrade-ns-system/rook.io-block...
2020-07-14 16:48:51.100063 I | ceph-cluster-controller: ConfigMap "rook-ceph-operator-config" changes detected. Updating configurations
2020-07-14 16:48:51.558862 I | ceph-cluster-controller: successfully started
2020-07-14 16:48:51.558935 I | ceph-cluster-controller: enabling hotplug orchestration
2020-07-14 16:48:51.558952 I | ceph-crashcollector-controller: successfully started
2020-07-14 16:48:51.559042 I | ceph-block-pool-controller: successfully started
2020-07-14 16:48:51.559113 I | ceph-object-store-user-controller: successfully started
2020-07-14 16:48:51.559162 I | ceph-object-realm-controller: successfully started
2020-07-14 16:48:51.559203 I | ceph-object-zonegroup-controller: successfully started
2020-07-14 16:48:51.559242 I | ceph-object-zone-controller: successfully started
2020-07-14 16:48:51.559376 I | ceph-object-controller: successfully started
2020-07-14 16:48:51.559434 I | ceph-file-controller: successfully started
2020-07-14 16:48:51.559482 I | ceph-nfs-controller: successfully started
2020-07-14 16:48:51.559588 I | operator: starting the controller-runtime manager
2020-07-14 16:48:51.880329 I | ceph-cluster-controller: reconciling ceph cluster in namespace "upgrade-ns"
2020-07-14 16:48:51.884061 I | op-mon: parsing mon endpoints: a=10.111.236.221:6789
2020-07-14 16:48:51.920645 I | op-mon: parsing mon endpoints: a=10.111.236.221:6789
2020-07-14 16:48:51.936024 I | op-agent: getting flexvolume dir path from FLEXVOLUME_DIR_PATH env var
2020-07-14 16:48:51.936044 I | op-agent: flexvolume dir path env var FLEXVOLUME_DIR_PATH is not provided. Defaulting to: /usr/libexec/kubernetes/kubelet-plugins/volume/exec/
2020-07-14 16:48:51.936072 I | op-agent: discovered flexvolume dir path from source default. value: /usr/libexec/kubernetes/kubelet-plugins/volume/exec/
2020-07-14 16:48:51.936080 I | op-agent: no agent mount security mode given, defaulting to 'Any' mode
2020-07-14 16:48:51.936087 W | op-agent: Invalid ROOK_ENABLE_SELINUX_RELABELING value "". Defaulting to "true".
2020-07-14 16:48:51.936093 W | op-agent: Invalid ROOK_ENABLE_FSGROUP value "". Defaulting to "true".
2020-07-14 16:48:51.944715 I | op-agent: rook-ceph-agent daemonset already exists, updating ...
2020-07-14 16:48:52.068794 I | op-k8sutil: ROOK_CSI_ENABLE_RBD="true" (configmap)
2020-07-14 16:48:52.131432 E | ceph-block-pool-controller: failed to reconcile failed to delete pool "upgradepool". : failed to list pools: failed to list pools: exit status 1
2020-07-14 16:48:52.134405 W | cephclient: failed to get ceph daemons versions, this likely means there is no cluster yet. failed to run 'ceph versions. Error initializing cluster client: ObjectNotFound('RADOS object not found (error calling conf_read_file)',)
: exit status 1
2020-07-14 16:48:52.253303 W | cephclient: failed to get ceph daemons versions, this likely means there is no cluster yet. failed to run 'ceph versions. Error initializing cluster client: ObjectNotFound('RADOS object not found (error calling conf_read_file)',)
: exit status 1
2020-07-14 16:48:52.253357 I | ceph-object-controller: reconciling object store deployments
2020-07-14 16:48:52.265225 I | op-k8sutil: ROOK_CSI_ENABLE_CEPHFS="true" (configmap)
2020-07-14 16:48:52.280196 E | ceph-file-controller: failed to reconcile failed to delete filesystem "upgrade-test-fs". : failed to get filesystem "upgrade-test-fs": failed to get file system upgrade-test-fs: exit status 1
2020-07-14 16:48:52.665493 I | op-k8sutil: ROOK_CSI_ALLOW_UNSUPPORTED_VERSION="false" (default)
2020-07-14 16:48:53.065489 I | op-k8sutil: ROOK_CSI_ENABLE_GRPC_METRICS="true" (configmap)
2020-07-14 16:48:53.272302 I | ceph-object-controller: ceph object store gateway service running at 10.101.142.210:80
2020-07-14 16:48:53.272786 I | ceph-object-controller: reconciling object store pools
2020-07-14 16:48:53.327075 E | ceph-block-pool-controller: failed to reconcile failed to delete pool "upgradepool". : failed to list pools: failed to list pools: exit status 1
2020-07-14 16:48:53.475909 I | op-k8sutil: ROOK_CSI_CEPH_IMAGE="quay.io/cephcsi/cephcsi:v2.1.2" (default)
2020-07-14 16:48:53.503488 W | ceph-object-controller: failed to adjust the PG count for rgw metadata pools. using the general default. failed to get config setting "rgw_rados_pool_pg_num_min" for user "mon.": exit status 1
2020-07-14 16:48:53.865782 I | op-k8sutil: ROOK_CSI_REGISTRAR_IMAGE="quay.io/k8scsi/csi-node-driver-registrar:v1.2.0" (default)
2020-07-14 16:48:53.937596 E | ceph-object-controller: failed to reconcile failed to create object store deployments: failed to create object pools: failed to create metadata pools: failed to create pool upgraded-object.rgw.control for object store upgraded-object.: failed to create crush rule upgraded-object.rgw.control: exit status 1
2020-07-14 16:48:54.065158 I | op-mon: parsing mon endpoints: a=10.111.236.221:6789
2020-07-14 16:48:54.159197 W | cephclient: failed to get ceph daemons versions, this likely means there is no cluster yet. failed to run 'ceph versions. Error initializing cluster client: ObjectNotFound('RADOS object not found (error calling conf_read_file)',)
: exit status 1
2020-07-14 16:48:54.253822 E | ceph-file-controller: failed to reconcile failed to delete filesystem "upgrade-test-fs". : failed to get filesystem "upgrade-test-fs": failed to get file system upgrade-test-fs: exit status 1
2020-07-14 16:48:54.265472 I | op-k8sutil: ROOK_CSI_PROVISIONER_IMAGE="quay.io/k8scsi/csi-provisioner:v1.6.0" (default)
2020-07-14 16:48:54.421186 E | ceph-block-pool-controller: failed to reconcile failed to delete pool "upgradepool". : failed to list pools: failed to list pools: exit status 1
2020-07-14 16:48:54.465361 I | op-k8sutil: ROOK_CSI_ATTACHER_IMAGE="quay.io/k8scsi/csi-attacher:v2.1.0" (default)
2020-07-14 16:48:54.666123 I | op-k8sutil: ROOK_CSI_SNAPSHOTTER_IMAGE="quay.io/k8scsi/csi-snapshotter:v2.1.1" (default)
2020-07-14 16:48:55.067676 I | op-k8sutil: ROOK_CSI_KUBELET_DIR_PATH="/var/lib/kubelet" (default)
2020-07-14 16:48:55.558830 E | ceph-block-pool-controller: failed to reconcile failed to delete pool "upgradepool". : failed to list pools: failed to list pools: exit status 1
2020-07-14 16:48:55.865038 I | op-mon: parsing mon endpoints: a=10.111.236.221:6789
2020-07-14 16:48:55.963828 W | cephclient: failed to get ceph daemons versions, this likely means there is no cluster yet. failed to run 'ceph versions. Error initializing cluster client: ObjectNotFound('RADOS object not found (error calling conf_read_file)',)
: exit status 1
2020-07-14 16:48:55.963872 I | ceph-object-controller: reconciling object store deployments
2020-07-14 16:48:56.266553 I | ceph-csi: successfully created csi config map "rook-ceph-csi-config"
2020-07-14 16:48:56.266815 I | ceph-csi: detecting the ceph csi image version for image "quay.io/cephcsi/cephcsi:v2.1.2"
2020-07-14 16:48:56.270498 I | ceph-cluster-controller: enabling ceph mon monitoring goroutine for cluster "upgrade-ns"
2020-07-14 16:48:56.270519 I | ceph-cluster-controller: enabling ceph osd monitoring goroutine for cluster "upgrade-ns"
2020-07-14 16:48:56.270532 I | ceph-cluster-controller: ceph status check interval is 5s
2020-07-14 16:48:56.270539 I | ceph-cluster-controller: enabling ceph status monitoring goroutine for cluster "upgrade-ns"
2020-07-14 16:48:56.270546 I | op-client: start watching client resources in namespace "upgrade-ns"
E0714 16:48:56.271595       6 reflector.go:153] pkg/mod/k8s.io/client-go@v0.17.2/tools/cache/reflector.go:105: Failed to list *v1.CephClient: the server could not find the requested resource (get cephclients.ceph.rook.io)
2020-07-14 16:48:56.369173 E | ceph-cluster-controller: failed to get ceph status. failed to get status. . Error initializing cluster client: ObjectNotFound('RADOS object not found (error calling conf_read_file)',): exit status 1
2020-07-14 16:48:56.465110 I | op-mon: parsing mon endpoints: a=10.111.236.221:6789
2020-07-14 16:48:56.563420 W | cephclient: failed to get ceph daemons versions, this likely means there is no cluster yet. failed to run 'ceph versions. Error initializing cluster client: ObjectNotFound('RADOS object not found (error calling conf_read_file)',)
: exit status 1
2020-07-14 16:48:56.734146 E | ceph-file-controller: failed to reconcile failed to delete filesystem "upgrade-test-fs". : failed to get filesystem "upgrade-test-fs": failed to get file system upgrade-test-fs: exit status 1
2020-07-14 16:48:56.745612 E | ceph-block-pool-controller: failed to reconcile failed to delete pool "upgradepool". : failed to list pools: failed to list pools: exit status 1
2020-07-14 16:48:57.065358 I | op-k8sutil: ROOK_OBC_WATCH_OPERATOR_NAMESPACE="false" (default)
2020-07-14 16:48:57.065379 I | op-bucket-prov: ceph bucket provisioner launched watching for provisioner "ceph.rook.io/bucket"
2020-07-14 16:48:57.066281 I | ceph-cluster-controller: detecting the ceph image version for image ceph/ceph:v14...
I0714 16:48:57.066387       6 manager.go:118] objectbucket.io/provisioner-manager "msg"="starting provisioner"  "name"="ceph.rook.io/bucket"
E0714 16:48:57.273544       6 reflector.go:153] pkg/mod/k8s.io/client-go@v0.17.2/tools/cache/reflector.go:105: Failed to list *v1.CephClient: the server could not find the requested resource (get cephclients.ceph.rook.io)
2020-07-14 16:48:57.947078 E | ceph-block-pool-controller: failed to reconcile failed to delete pool "upgradepool". : failed to list pools: failed to list pools: exit status 1
2020-07-14 16:48:58.067041 I | ceph-object-controller: ceph object store gateway service running at 10.101.142.210:80
2020-07-14 16:48:58.067060 I | ceph-object-controller: reconciling object store pools
E0714 16:48:58.274717       6 reflector.go:153] pkg/mod/k8s.io/client-go@v0.17.2/tools/cache/reflector.go:105: Failed to list *v1.CephClient: the server could not find the requested resource (get cephclients.ceph.rook.io)
2020-07-14 16:48:58.274923 W | ceph-object-controller: failed to adjust the PG count for rgw metadata pools. using the general default. failed to get config setting "rgw_rados_pool_pg_num_min" for user "mon.": exit status 1
2020-07-14 16:48:58.668386 I | op-mon: parsing mon endpoints: a=10.111.236.221:6789
2020-07-14 16:48:58.809057 E | ceph-object-controller: failed to reconcile failed to create object store deployments: failed to create object pools: failed to create metadata pools: failed to create pool upgraded-object.rgw.control for object store upgraded-object.: failed to create crush rule upgraded-object.rgw.control: exit status 1
2020-07-14 16:48:59.033935 W | cephclient: failed to get ceph daemons versions, this likely means there is no cluster yet. failed to run 'ceph versions. Error initializing cluster client: ObjectNotFound('RADOS object not found (error calling conf_read_file)',)
: exit status 1
2020-07-14 16:48:59.172360 E | ceph-block-pool-controller: failed to reconcile failed to delete pool "upgradepool". : failed to list pools: failed to list pools: exit status 1
2020-07-14 16:48:59.256652 E | ceph-file-controller: failed to reconcile failed to delete filesystem "upgrade-test-fs". : failed to get filesystem "upgrade-test-fs": failed to get file system upgrade-test-fs: exit status 1
E0714 16:48:59.275889       6 reflector.go:153] pkg/mod/k8s.io/client-go@v0.17.2/tools/cache/reflector.go:105: Failed to list *v1.CephClient: the server could not find the requested resource (get cephclients.ceph.rook.io)
2020-07-14 16:48:59.815597 I | op-mon: parsing mon endpoints: a=10.111.236.221:6789
2020-07-14 16:49:00.218927 W | cephclient: failed to get ceph daemons versions, this likely means there is no cluster yet. failed to run 'ceph versions. Error initializing cluster client: ObjectNotFound('RADOS object not found (error calling conf_read_file)',)
: exit status 1
2020-07-14 16:49:00.218973 I | ceph-object-controller: reconciling object store deployments
E0714 16:49:00.280130       6 reflector.go:153] pkg/mod/k8s.io/client-go@v0.17.2/tools/cache/reflector.go:105: Failed to list *v1.CephClient: the server could not find the requested resource (get cephclients.ceph.rook.io)
2020-07-14 16:49:00.399460 I | op-mon: parsing mon endpoints: a=10.111.236.221:6789
2020-07-14 16:49:00.613585 E | ceph-block-pool-controller: failed to reconcile failed to delete pool "upgradepool". : failed to list pools: failed to list pools: exit status 1
2020-07-14 16:49:00.697716 I | ceph-csi: Detected ceph CSI image version: "v2.1.2"
2020-07-14 16:49:00.752321 W | cephclient: failed to get ceph daemons versions, this likely means there is no cluster yet. failed to run 'ceph versions. Error initializing cluster client: ObjectNotFound('RADOS object not found (error calling conf_read_file)',)
: exit status 1
2020-07-14 16:49:00.884065 I | ceph-object-controller: ceph object store gateway service running at 10.101.142.210:80
2020-07-14 16:49:00.884086 I | ceph-object-controller: reconciling object store pools
2020-07-14 16:49:01.070876 I | op-k8sutil: CSI_FORCE_CEPHFS_KERNEL_CLIENT="true" (default)
2020-07-14 16:49:01.077217 E | ceph-file-controller: failed to reconcile failed to delete filesystem "upgrade-test-fs". : failed to get filesystem "upgrade-test-fs": failed to get file system upgrade-test-fs: exit status 1
2020-07-14 16:49:01.214279 W | ceph-object-controller: failed to adjust the PG count for rgw metadata pools. using the general default. failed to get config setting "rgw_rados_pool_pg_num_min" for user "mon.": exit status 1
2020-07-14 16:49:01.269744 I | op-k8sutil: CSI_CEPHFS_GRPC_METRICS_PORT="9091" (default)
E0714 16:49:01.283898       6 reflector.go:153] pkg/mod/k8s.io/client-go@v0.17.2/tools/cache/reflector.go:105: Failed to list *v1.CephClient: the server could not find the requested resource (get cephclients.ceph.rook.io)
2020-07-14 16:49:01.479549 I | op-k8sutil: CSI_CEPHFS_LIVENESS_METRICS_PORT="9081" (default)
2020-07-14 16:49:01.669009 I | op-k8sutil: CSI_RBD_GRPC_METRICS_PORT="9090" (default)
2020-07-14 16:49:01.759805 E | ceph-cluster-controller: failed to get ceph status. failed to get status. . Error initializing cluster client: ObjectNotFound('RADOS object not found (error calling conf_read_file)',): exit status 1
2020-07-14 16:49:01.871630 I | op-k8sutil: CSI_RBD_LIVENESS_METRICS_PORT="9080" (default)
2020-07-14 16:49:01.920321 E | ceph-object-controller: failed to reconcile failed to create object store deployments: failed to create object pools: failed to create metadata pools: failed to create pool upgraded-object.rgw.control for object store upgraded-object.: failed to create crush rule upgraded-object.rgw.control: exit status 1
2020-07-14 16:49:01.998952 E | ceph-block-pool-controller: failed to reconcile failed to delete pool "upgradepool". : failed to list pools: failed to list pools: exit status 1
2020-07-14 16:49:02.065649 I | op-k8sutil: CSI_PLUGIN_PRIORITY_CLASSNAME="" (default)
E0714 16:49:02.287776       6 reflector.go:153] pkg/mod/k8s.io/client-go@v0.17.2/tools/cache/reflector.go:105: Failed to list *v1.CephClient: the server could not find the requested resource (get cephclients.ceph.rook.io)
2020-07-14 16:49:02.465413 I | op-k8sutil: CSI_PROVISIONER_PRIORITY_CLASSNAME="" (default)
2020-07-14 16:49:02.867580 I | ceph-cluster-controller: detected ceph image version: "14.2.10-0 nautilus"
2020-07-14 16:49:02.867606 I | ceph-cluster-controller: validating ceph version from provided image
2020-07-14 16:49:03.065302 I | op-k8sutil: CSI_CEPHFS_PLUGIN_UPDATE_STRATEGY="RollingUpdate" (default)
2020-07-14 16:49:03.267949 I | op-mon: parsing mon endpoints: a=10.111.236.221:6789
E0714 16:49:03.293984       6 reflector.go:153] pkg/mod/k8s.io/client-go@v0.17.2/tools/cache/reflector.go:105: Failed to list *v1.CephClient: the server could not find the requested resource (get cephclients.ceph.rook.io)
2020-07-14 16:49:03.657936 E | ceph-block-pool-controller: failed to reconcile failed to delete pool "upgradepool". : failed to list pools: failed to list pools: exit status 1
2020-07-14 16:49:03.719495 W | cephclient: failed to get ceph daemons versions, this likely means there is no cluster yet. failed to run 'ceph versions. Error initializing cluster client: ObjectNotFound('RADOS object not found (error calling conf_read_file)',)
: exit status 1
2020-07-14 16:49:03.955809 E | ceph-file-controller: failed to reconcile failed to delete filesystem "upgrade-test-fs". : failed to get filesystem "upgrade-test-fs": failed to get file system upgrade-test-fs: exit status 1
2020-07-14 16:49:04.265124 I | op-k8sutil: CSI_RBD_PLUGIN_UPDATE_STRATEGY="RollingUpdate" (default)
E0714 16:49:04.294921       6 reflector.go:153] pkg/mod/k8s.io/client-go@v0.17.2/tools/cache/reflector.go:105: Failed to list *v1.CephClient: the server could not find the requested resource (get cephclients.ceph.rook.io)
2020-07-14 16:49:04.465062 I | op-mon: parsing mon endpoints: a=10.111.236.221:6789
2020-07-14 16:49:04.465447 I | cephconfig: writing config file /var/lib/rook/upgrade-ns/upgrade-ns.config
2020-07-14 16:49:04.465584 I | cephconfig: generated admin config in /var/lib/rook/upgrade-ns
2020-07-14 16:49:04.671025 I | op-mon: parsing mon endpoints: a=10.111.236.221:6789
2020-07-14 16:49:04.872752 I | op-k8sutil: ROOK_CSI_RESIZER_IMAGE="quay.io/k8scsi/csi-resizer:v0.4.0" (default)
2020-07-14 16:49:05.067471 I | op-k8sutil: CSI_LOG_LEVEL="" (default)
2020-07-14 16:49:05.075560 I | ceph-csi: successfully started CSI Ceph RBD
2020-07-14 16:49:05.077808 I | ceph-csi: successfully started CSI CephFS driver
2020-07-14 16:49:05.116930 I | ceph-cluster-controller: cluster "upgrade-ns": version "14.2.10-0 nautilus" detected for image "ceph/ceph:v14"
E0714 16:49:05.296006       6 reflector.go:153] pkg/mod/k8s.io/client-go@v0.17.2/tools/cache/reflector.go:105: Failed to list *v1.CephClient: the server could not find the requested resource (get cephclients.ceph.rook.io)
2020-07-14 16:49:05.343885 I | ceph-object-controller: reconciling object store deployments
2020-07-14 16:49:05.465325 I | op-k8sutil: CSI_PROVISIONER_TOLERATIONS="" (default)
2020-07-14 16:49:05.665551 I | op-mon: parsing mon endpoints: a=10.111.236.221:6789
2020-07-14 16:49:06.066352 I | op-k8sutil: CSI_PROVISIONER_NODE_AFFINITY="" (default)
2020-07-14 16:49:06.159040 I | op-config: CephCluster "upgrade-ns" status: "Progressing". "Cluster is checking if updates are needed"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x58 pc=0x18677eb]

goroutine 705 [running]:
github.com/rook/rook/pkg/operator/ceph/cluster/mon.(*Cluster).checkHealth(0xc0010e3400, 0x0, 0x0)
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/cluster/mon/health.go:104 +0xab
github.com/rook/rook/pkg/operator/ceph/cluster/mon.(*HealthChecker).Check(0xc000f5b720, 0xc000cc8cc0)
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/cluster/mon/health.go:92 +0x143
created by github.com/rook/rook/pkg/operator/ceph/cluster.(*ClusterController).startMonitoringCheck
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/cluster/monitoring.go:116 +0x683
```
* Crashing pod(s) logs, if necessary

 To get logs, use `kubectl -n <namespace> logs <pod name>`
When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
Read [Github documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).

**Environment**:
* OS (e.g. from /etc/os-release):
* Kernel (e.g. `uname -a`):
* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod):
* Storage backend version (e.g. for ceph do `ceph -v`):
* Kubernetes version (use `kubectl version`):
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)):

**Comment by BlaineEXE** [comment]
It looks like this might be related to changes from https://github.com/rook/rook/pull/5789

**Comment by BlaineEXE** [comment]
First attempt to determine the source of the issue is an attempted fix here: https://github.com/rook/rook/pull/4916/commits/7716a999d4bb7a6319281caa755836e730a48f75

**Comment by BlaineEXE** [comment]
This might have been fixed by #5746. Setting do-not-merge for now to see if this issue persists or is fixed now.

**Comment by BlaineEXE** [comment]
@travisn can we close this as confirmed fixed by one of your recent PRs?

**Comment by travisn** [comment]
> @travisn can we close this as confirmed fixed by one of your recent PRs?

Yes, this is expected to be fixed by #5869.
